src/nvim/os/fs.c

```diff
@@ -1020,6 +1023,7 @@ bool os_fileinfo(const char *path, FileInfo *file_info)
 bool os_fileinfo_link(const char *path, FileInfo *file_info)
   FUNC_ATTR_NONNULL_ARG(2)
 {
+  memset(file_info, 0, sizeof(*file_info));
   if (path == NULL) {
     return false;
   }
```

Comment:
No need to call `file_info->stat = request.statbuf;` depending on result ?

Reply:
Good catch.  Thanks!